### PR TITLE
Add reverse CI for oneAPI and AMDGPU

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -54,6 +54,59 @@ steps:
...
+          codecov: true
+    command: |
+      julia -e 'println("--- :julia: Instantiating project")
+                using Pkg
+                Pkg.develop(; path=pwd())
+                Pkg.develop(; name="AMDGPU")' || exit 3
+
+      julia -e 'println("+++ :julia: Running tests")
+                using Pkg
+                Pkg.test("AMDGPU"; coverage=true)'
```
Comment (vchuravy) [inline .buildkite/pipeline.yml:102]: @pxl-th @jpsamaroo is there a way to run only a certain test?

Comment (pxl-th) [reply .buildkite/pipeline.yml:102]: Not at the moment :/

**Review by maleadt — APPROVED** [review]
LGTM.

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -54,6 +54,59 @@ steps:
...
+      julia -e 'println("--- :julia: Instantiating project")
+                using Pkg
+                Pkg.develop(; path=pwd())
+                Pkg.develop(; name="oneAPI")' || exit 3
+
+      julia -e 'println("+++ :julia: Running tests")
+                using Pkg
+                Pkg.test("oneAPI"; coverage=true, test_args=["kernelabstractions"])'
+    agents:
+      queue: "juliaecosystem"
```
Comment (maleadt) [inline .buildkite/pipeline.yml:77]: oneAPI runners aren't on the juliaecosystem queue yet, but on the juliagpu one.

**Review by maleadt — APPROVED** [review]

**Pull request merged**